### ISSUE TITLE
[4.0] Modern Router URL #SEO issue on upgrade?

### ISSUE THREAD

**Issue by Ruud68** [joomla/joomla-cms#32923]
### Steps to reproduce the issue
setup (or use) a Joomla 3.9 / 3.10 site
create nested categories e.g News > TSX Venture > NM
create article in the NM category e.g. test article
setup menu to main category News > type category blog
set article options > integration > legacy router

1] on front-end go to menu test, click the tsx venture category > click the nm category > open test article > note the URL

upgrade 3.9 to 3.10 to 4.0
you want to retain your URL for SEO purposes so make sure that you have turned OFF 'Remove ID's from URLs' in article options

2] on front-end go to menu test, click the tsx venture category > click the nm category > open test article note the URL
 
### Expected result

both URLs should be the same

### Actual result

URL 1]: http://localhost/test/310/test/12-tsx-venture/nm/7-test-article
URL 2]: http://localhost/test/310/test/11-tsx-venture/12-nm/7-test-article

### Additional comments

Upgrading existing J3 sites (with legacy router) will break B\C on the URL build with J4 (modern router) with Article IDs

**Comment by wilsonge** [comment]
This is expected behaviour. The entire point of the new router is to fix duplicate URLs to the same content. Yes it is expected this means that previously functional urls will 404. Either way this is fully described in https://github.com/joomla/joomla-cms/issues/14848

**Comment by Ruud68** [comment]
Hi @wilsonge thanks for the follow up, totally forgot about that issue. Just see that I participated in that discussion and also offered a solution I created for it, for which I btw never got any response / feedback).
I understand that this is / was 'expected' behavior at the time it was discussed, but since then the world 'has moved on' and abusers are targeting this 'feature'.
So IMO this should be addressed with the new knowledge we have now, it is like fixing a security issue that was already present in version 3.2 at the time 3.2 was released you had no knowledge of the issue, now you do...
